src/main/java/net/minecraftforge/fml/config/ConfigFileTypeHandler.java

```diff
@@ -91,6 +92,38 @@ private boolean setupConfigFile(final ModConfig modConfig, final Path file, fina
         return true;
     }
 
+    public static void backUpConfig(final CommentedFileConfig commentedFileConfig)
+    {
+        backUpConfig(commentedFileConfig, 5); //TODO: Think of a way for mods to set their own preference (include a sanity check as well, no disk stuffing)
+    }
+
+    public static void backUpConfig(final CommentedFileConfig commentedFileConfig, final int maxBackups)
+    {
+        Path bakFileLocation = commentedFileConfig.getNioPath().getParent();
+        String bakFileName = FilenameUtils.removeExtension(commentedFileConfig.getFile().getName());
+        String bakFileExtension = FilenameUtils.getExtension(commentedFileConfig.getFile().getName()) + ".bak";
+        Path bakFile = bakFileLocation.resolve(bakFileName + "-1" + "." + bakFileExtension);
+        try
+        {
+            for(int i = maxBackups; i > 0; i--)
+            {
+                Path oldBak = bakFileLocation.resolve(bakFileName + "-" + i + "." + bakFileExtension);
+                if(Files.exists(oldBak))
+                {
+                    if(i >= maxBackups)
+                        Files.delete(oldBak);
+                    else
+                        Files.move(oldBak, bakFileLocation.resolve(bakFileName + "-" + (i + 1) + "." + bakFileExtension));
+                }
+            }
+            Files.copy(commentedFileConfig.getNioPath(), bakFile);
+        }
+        catch (IOException exception)
+        {
+            LOGGER.warn(CONFIG, "Failed to back up config file " + commentedFileConfig.getNioPath(), exception);
```

Comment:
Before anyone comments on this part, Log4J doesn't seem to have a method for both {} params and throwable printing at the same time.

Reply:
Actually it does, I looked it up at one point, and it turns out log4j has "hidden" functionality that makes it treat it as such if it has one more passed to it than the parameters and is a throwable.

Reply:
TIL